src/virt_card.py

```diff
@@ -97,13 +100,22 @@ def run_cmd(self, cmd: str = None, expect: str = None, pin: bool = True, passwd:
                 shell.sendline(passwd)
 
             if expect is not None:
-                out = shell.expect([pexpect.TIMEOUT, expect], timeout=20)
-
-                if out != 1:
-                    if out == 0:
-                        log.error("Time out")
-                    raise pexpect.exceptions.EOF(f"Pattern '{expect}' is not "
-                                                 f"found in the output.")
+                if reject is not None:
+                    out = shell.expect([reject, pexpect.TIMEOUT, expect], timeout=20)
+                    if out != 2:
+                        if out == 1:
+                            log.error("Time out")
+                        if out == 0;
+                            log.error("Disallowed pattern found")
+                        raise pexpect.exceptions.EOF(f"Pattern '{reject}' is "
+                                                     f"found in the output.")
+                else:
```

Comment:
Hm, I think we still want to match expected pattern. I would suggest to leave this section as it was without else branch. We have to patterns: the one that we realy **want to find** (*expect*) and one that we don't want to have in the output (*reject*). So, if these two patterns are set, we want to check both. In this implementation with two patterns set only *reject* pattern would be match

Reply:
Current implementation try to match expected pattern bud send log.error and skip matching the pattern in case it hits undesired pattern. The above suggestion would run the same command twice if the 'reject' pattern is defined. I don't think it's good solution. Let's discuss this one.